cnd/src/swap_protocols/rfc003/bitcoin/htlc_events.rs

```diff
@@ -68,53 +80,50 @@ impl HtlcEvents<Bitcoin, Amount> for Arc<dyn QueryBitcoin + Send + Sync> {
         &self,
         htlc_params: HtlcParams<Bitcoin, Amount>,
         htlc_deployment: &Deployed<Bitcoin>,
-        _: &Funded<Bitcoin, Amount>,
+        _htlc_funding: &Funded<Bitcoin, Amount>,
     ) -> Box<RedeemedOrRefundedFuture<Bitcoin>> {
         let refunded_future = {
-            let query_bitcoin = Arc::clone(&self);
-            let id = htlc_params.query_id_refunded();
-            let refunded_query = self
-                .create(&id, BitcoinQuery::refund_htlc(htlc_deployment.location))
-                .inspect(|query_id| log::debug!("Refund query id {:?}", query_id))
-                .map_err(rfc003::Error::Btsieve);
-
-            refunded_query
-                .and_then(move |query_id| {
-                    query_bitcoin
-                        .transaction_first_result(&query_id)
-                        .map_err(rfc003::Error::Btsieve)
-                })
-                .map(Refunded::<Bitcoin>::new)
+            self.matching_transactions(TransactionQuery {
+                to_address: None,
+                from_outpoint: Some(htlc_deployment.location),
+                unlock_script: Some(vec![vec![]]),
+            })
+            .map_err(|_| rfc003::Error::Btsieve)
+            .first_or_else(|| {
+                log::warn!("stream of matching transactions ended before yielding a value");
+                rfc003::Error::Btsieve
```

Comment:
Didn't you say either log it _or_ return it ;)

Reply:
Since the error variant does not contain the information of what went wrong, not logging here would be a loss of information :)

Reply:
I stand corrected.  Thanks.